services/app/src/components/modal/UserModal.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import Modal from 'react-modal';
+
+const customStyles = {
+  content: {
+    top: '50%',
+    left: '50%',
+    right: 'auto',
+    bottom: 'auto',
+    marginRight: '-50%',
+    transform: 'translate(-50%, -50%)',
+  },
+};
+const UserModal = () => {
+  const [modalIsOpen, setIsOpen] = useState(false);
+
+  const closeModal = () => {
+    setIsOpen(false);
+  };
+
+  const submitInfo = (event) => {
+    event.preventDefault();
+    window.localStorage.setItem('user', event.target.username.value);
+    window.location.reload();
```

Comment:
Reloading an application on login does not seem like a right approach, as that would force user to re-download all the javascript after logging in.

I do understand, that this is needed to redirect the user to game screen, but there should be a better way. I would suggest to introduce additional route for `LoginScreen` that would contain the `UserModal` and unauthenticated users would be forced to this `LoginScreen`. Something like this should do the trick

```
const AuthenticatedRoute = (routeProps) => {
  const { authentication } = useUserFromLocalStorage();
  // set requested url as redirect URL on localStorage on context
  if (authentication) {
    return <Route {...routeProps} />;
  }

  return <Redirect to="/login" />
}

function Content() {
  return (
    <Routes>
      <Route path="/login" element={<LoginScreen />} />
      <AuthenticatedRoute path="/game/:id" element={<Game />} />
    </Routes>
  );
}
```

With this, you should be able to use from `UserModal/LoginScreen`
```
import { useNavigate } from 'react-router-dom';
<...>
const navigate = useNavigate();
const redirectUrl = // get redirect URL from localStorage or context
const submitHandler = () => {
  <...>
  navigate(redirectUrl)
}
```